manager/controllers/app/select_modules.go

```diff
@@ -230,9 +240,13 @@ func (p *PathBuilder) findPathsWithinLimit(source, sink *Node, n int) []Solution
 			if !supportsSinkInterface(&edge, sink) {
 				continue
 			}
+
 			edge.Sink = sink
 			// if a module supports both source and sink interfaces, it's an end of the recursion
 			if supportsSourceInterface(&edge, source) {
+				if sink.Connection.DataFormat == "" {
```

Comment:
Why here? It should be done at the end when the path has been validated.

Reply:
Fixed. Thanks